Src/BootCamp.Chapter/Inventory.cs

```diff
@@ -15,12 +16,75 @@ public Inventory()
 
         public Item[] GetItems(string name)
         {
-            return new Item[0];
+            var foundItems = new Item[0];
+            foreach (var item in _items)
+            {
+                if (item.GetName() == name)
+                {
+                    foundItems = Add(item);
+                }
+            }
+
+            return foundItems;
         }
 
         public void AddItem(Item item)
         {
+            _items = Add(item);
+        }
+
+        public Item[] Add(Item item)
+        {
+            var newArray = new Item[_items.Length + 1];
+            for (int i = 0; i <= _items.Length - 1; i++)
+            {
+                newArray[i] = _items[i];
+            }
+            newArray[^1] = item;
+            return newArray;
+        }
 
+        public bool SearchItem(string name)
+        {
+            foreach (var item in _items)
+            {
+                if (item.GetName() == name)
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        public void RemoveByName(string name)
+        {
+            _items = RemoveByBame(name);
+        }
+
+        public Item[] RemoveByBame(string name)
```

Comment:
:D

Reply:
sorry, I do not understand what you mean with the emonicon.
But the homework is approved so im happy

Reply:
Remove by bame is a funny typo

Reply:
oke, never noticed that one